File: app/src/main/java/spiral/bit/dev/sunshinenotes/data/CheckListDAO.java
package spiral.bit.dev.sunshinenotes.data;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import spiral.bit.dev.sunshinenotes.models.Task;
import spiral.bit.dev.sunshinenotes.models.CheckList;

@Dao
public interface CheckListDAO {

    @Query("SELECT * FROM check_lists ORDER BY check_list_id DESC")
    List<CheckList> getAllCheckLists();

    @Query("SELECT * FROM check_items WHERE parent_id = :parentId")
    List<Task> getAllTasks(int parentId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertCheckList(CheckList checkList);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTask(Task task);

    @Query("DELETE FROM check_lists WHERE check_list_id = :id")
    void deleteCheckListById(int id);

        @Delete
    void deleteCheckList(CheckList checkList);

    @Delete
    void deleteTask(Task task);

    @Query("DELETE FROM check_items WHERE parent_id = :parentId")
    void deleteCheckById(int parentId);
}

File: app/src/main/java/spiral/bit/dev/sunshinenotes/data/CheckListDAO.kt
package spiral.bit.dev.sunshinenotes.data

import androidx.room.*
import spiral.bit.dev.sunshinenotes.models.CheckList
import spiral.bit.dev.sunshinenotes.models.Task

@Dao
interface CheckListDAO {
    @get:Query("SELECT * FROM check_lists ORDER BY check_list_id DESC")
    val allCheckLists: List<CheckList?>?

    @Query("SELECT * FROM check_items WHERE parent_id = :parentId")
    fun getAllTasks(parentId: Int): List<Task?>?

    @Delete
    fun deleteMultipleCheckLists(checkList: List<CheckList>)

    @Delete
    fun deleteMultipleTasks(tasks: List<Task>)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertCheckList(checkList: CheckList?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTask(task: Task?)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateTask(task: Task?)

    @Query("DELETE FROM check_lists WHERE check_list_id = :id")
    fun deleteCheckListById(id: Int)

    @Delete
    fun deleteCheckList(checkList: CheckList?)

    @Delete
    fun deleteTask(task: Task?)

    @Query("DELETE FROM check_items WHERE parent_id = :parentId")
    fun deleteCheckById(parentId: Int)

    @Query("DELETE FROM check_items WHERE id = :id")
    fun deleteCheckItemById(id: Int)
}